yt/geometry/oct_container.pyx

```diff
@@ -581,6 +595,12 @@ cdef class OctreeContainer:
         cdef int in_boundary = 0
         # How do we bootstrap ourselves?
         for p in range(no):
+            # We allow specifying curlevel = -1 to query from the levels array
+            # instead.
+            if curlevel == -1:
+                this_level = levels[p]
+            else:
+                this_level = curlevel
```

Comment:
this doesn't depend on the iteration variable (`p`), so it would be clearer (and more efficient) to put it outside the loop

Reply:
`levels[p]`

Reply:
🤦🏻‍♂️